django/db/models/base.py

```diff
@@ -834,6 +834,11 @@ def save_base(self, raw=False, force_insert=False,
         # Once saved, this is no longer a to-be-added instance.
         self._state.adding = False
 
+        # Readonly fields are computed by the Database. If there are any,
+        # we should fetch their value. (refresh_from_db does nothing if the
+        # field list is empty)
+        self.refresh_from_db(fields=[field.name for field in self._meta.readonly_fields])
```

Comment:
I've thought about this more and think this should be disabled by default (or even removed, users can manually call `refresh_from_db` after `save`). I think most use cases for readonly fields, python code will not require the new value from the database immediately. Most Django workflows typically take user data and update the database, and return - not requiring the new data from the database. In addition, virtual columns will be a key use case for readonly fields, but they are mostly added just for allowing more selectivity on queries. By adding an extra query to every single `save()` on any model with a readonly field, we could be introducing a major performance regression for some users, for the off chance use case that after `save()`ing a model, the readonly columns' values are required. Even if it is kept, it should be flagged, so that users can avoid the performance hit.